### PR TITLE
Some notes for discussion

### PR DESCRIPTION
**Pull request by hadley**

### CODE REVIEW

File: R/utils-github.R [hunk 1]
```diff
@@ -171,67 +171,103 @@ github_remote_list <- function(these = c("origin", "upstream"), x = NULL) {
...
+        c(
+          "Can't reach {.url {api_url}}.",
+          "Are you offline? Is GitHub down?"
+        ),
+        parent = cnd,
+        call = error_call
+      )
+    },
+    httr2_http_404 = function(cnd) {
+      # Might be because it's a private repo & no PAT
```
Comment (hadley) [inline R/utils-github.R:251]: https://docs.github.com/en/rest/using-the-rest-api/troubleshooting-the-rest-api?apiVersion=2022-11-28#404-not-found-for-an-existing-resource

File: R/utils-github.R [hunk 2]
```diff
@@ -171,67 +171,103 @@ github_remote_list <- function(these = c("origin", "upstream"), x = NULL) {
...
+            "Do you need to configure a GitHub token?",
+            "See {.run usethis::gh_token_help()} for advice."
+          )
+        ),
+        parent = cnd,
+        call = error_call
+      )
+    },
+    httr2_http_403 = function(cnd) {
+      # No token?
```
Comment (hadley) [inline R/utils-github.R:265]: I don't _think_ it's possible to get this error?

**Comment by jennybc** [comment]
Not a full review but glancing at this reminded me: do we need to use "http" here instead of "httr2"? If you look around, that's what's done in similar situations already. E.g. "http_error_403". Didn't we learn that's how gh classes things?

File: R/utils-github.R (hunk 2 above)
Comment (hadley) [reply R/utils-github.R:265]: Nope, see https://github.com/r-lib/usethis/issues/1929

**Comment by hadley** [comment]
Giving up on this approach

**Pull request closed without merging**